arch/arm64/core/reset.c

```diff
@@ -88,8 +88,11 @@ void z_arm64_el3_init(void)
 	reg |= (SCR_RES1 |		/* RES1 */
 		SCR_RW_BIT |		/* EL2 execution state is AArch64 */
 		SCR_ST_BIT |		/* Do not trap EL1 accesses to timer */
-		SCR_HCE_BIT |		/* Do not trap HVC */
-		SCR_SMD_BIT);		/* Do not trap SMC */
+		SCR_HCE_BIT);		/* Do not trap HVC */
+#ifndef CONFIG_ARM64_ENABLE_EL3_SERVICE
+	reg |= SCR_SMD_BIT;		/* Do not trap SMC */
```

Comment:
Shouldn't we set `SCR_EL3.SMD` to `0` instead?

Reply:
SCR_EL3.SMD is being set to 1 if feature is not enabled (ifndef).  If feature is enabled SCR_EL3.SMD will be 0 due to reg initially being set to 0.

Reply:
oh right, I didn't see that was an `ifndef`